uv-tutorial.md

```diff
@@ -0,0 +1,98 @@
+# Tutorial for using the uv package manager
+
+[uv](https://docs.astral.sh/uv/) is a Python package manager meant to act as a drop-in
+replacement for `pip` (and many more tools). In this project, we use it to manage our
+packages, similar to how `poetry` works. This is done using a lockfile called
+`uv.lock`.
+
+## uv as a drop-in replacement for pip
+
+`uv` is a lot faster than `pip`, so we recommend installing packages from `PyPI`
+with `uv pip install <package>` instead of `pip install <package>`. You don't need to
+change anything in your project to use this feature, as it works as a drop-in
+replacement to `pip`.
+
+## uv as a project-wide package manager
+
+If you wish to use the `uv sync` and/or `uv lock` commands, which is how you use `uv`
+to manage all your project packages, then note that these commands will only work
+with the directory called `.venv` in the project directory. Sometimes, this can be a
+bit annoying, especially with an existing venv, so we recommend using a
+[symlink](https://en.wikipedia.org/wiki/Symbolic_link).
+
+### Symlinking .venv
+
+To create a symlink between your venv and the `.venv` directory, you can use the
+following command:
+
+```bash
+ln -s <path/to/your_venv> <path/to/.venv>
+```
+
+As an example, if I am in the `itwinai/` folder and my venv is called `envAI_juwels`,
+then the following will create the wanted symlink:
+
+```bash
+ln -s envAI_juwels .venv
+```
+
+### Installing from uv.lock
+
+> [!Warning]
+> If `uv` creates your venv for you, the venv will not contain `pip`. However, you need
+> to have `pip` installed to be able to run the installation scripts for `Horovod` and
+> `DeepSpeed`, so we have included `pip` in the dependencies in `pyproject.toml`.
+
+To install from the `uv.lock` file into the `.venv` venv, you can do the following:
+
+```bash
+uv sync
+```
+
+If the `uv.lock` file has optional dependencies (e.g. `macos` or `torch`), then these
+can be added with the `--extra` flag as follows:
+
+```bash
+uv sync --extra torch --extra macos
```

Comment:
This is cool. So basically we can have a single lock file which locks both macos and linux dependencies in the same place, right?

Reply:
That is how I understand it, yeah 😎

Reply:
cool